components/brave_wallet_ui/common/async/lib.ts

```diff
@@ -286,3 +286,12 @@ export function refreshSitePermissions () {
     dispatch(WalletActions.setSitePermissions({ accounts: accountsWithPermission }))
   }
 }
+
+export async function hardwareDeviceIdFromAddress (address: string) {
+  const utf8 = new TextEncoder()
+  const msgBuffer = utf8.encode(address)
+  const hashBuffer = await Promise.resolve(crypto.subtle.digest('SHA-256', msgBuffer))
+  const hashArray = Array.from(new Uint8Array(hashBuffer))
+  const hashHex = hashArray.map(b => b.toString(16).padStart(2, '0')).join('')
+  return hashHex
+}
```

Comment:
This function should not be here. This module (and directory) are about dispatching and handling actions with the redux store. Suggest just making a simple module at `common/hardwareDeviceIdFromAddress.ts`

Reply:
And this doesn't look right:
`await Promise.resolve(crypto.subtle.digest('SHA-256', msgBuffer))` is exactly the same as just writing `crypto.subtle.digest('SHA-256', msgBuffer)`, so this function doesn't need to be `async` at all.

Reply:
moved to new file but keep old version, it throws errors otherwise:
```
      TS2769: No overload matches this call.
  Overload 1 of 5, '(elements: Iterable<number>): Uint8Array', gave the following error.
    Argument of type 'PromiseLike<ArrayBuffer>' is not assignable to parameter of type 'Iterable<number>'.
      Property '[Symbol.iterator]' is missing in type 'PromiseLike<ArrayBuffer>' but required in type 'Iterable<number>'.
  Overload 2 of 5, '(length: number): Uint8Array', gave the following error.
    Argument of type 'PromiseLike<ArrayBuffer>' is not assignable to parameter of type 'number'.
  Overload 3 of 5, '(arrayOrArrayBuffer: ArrayLike<number> | ArrayBuffer | SharedArrayBuffer): Uint8Array', gave the following error.
    Argument of type 'PromiseLike<ArrayBuffer>' is not assignable to parameter of type 'ArrayLike<number> | ArrayBuffer | SharedArrayBuffer'.
      Type 'PromiseLike<ArrayBuffer>' is missing the following properties from type 'SharedArrayBuffer': byteLength, length, slice, [Symbol.species], [Symbol.toStringTag]
```